File: 品味Java的21種設計模式/dp_cpp/visitor/example4/PersonalCustomer.cpp
#include "PersonalCustomer.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace visitor
			{
				namespace example4
				{

					std::string PersonalCustomer::getTelephone()
					{
						return telephone;
					}

					void PersonalCustomer::setTelephone(std::string telephone)
					{
						this->telephone = telephone;
					}

					int PersonalCustomer::getAge()
					{
						return age;
					}

					void PersonalCustomer::setAge(int age)
					{
						this->age = age;
					}

					void PersonalCustomer::accept(Visitor *visitor)
					{
						//回调访问者对象的相应方法
						visitor->visitPersonalCustomer(this);
					}
				}
			}
		}
	}
}

File: 品味Java的21種設計模式/dp_java/visitor/example4/PersonalCustomer.java
package cn.javass.dp.visitor.example4;
/**
 * 个人客户
 */
public class PersonalCustomer extends Customer {
    /**
     * 联系电话
     */
    private String telephone;
    /**
     * 年龄
     */
    private int age;

    public String getTelephone() {
        return telephone;
    }
    public void setTelephone(String telephone) {
        this.telephone = telephone;
    }
    public int getAge() {
        return age;
    }
    public void setAge(int age) {
        this.age = age;
    }
    public void accept(Visitor visitor) {
        //回调访问者对象的相应方法
        visitor.visitPersonalCustomer(this);
    }
}